components/net_connect/connect.c

```diff
@@ -0,0 +1,189 @@
+            }
+#endif
+        }
+    }
+    return ESP_OK;
+}
+
+void net_connect_print_all_netif_ips(const char *prefix)
+{
+    // Print all IPs in TCPIP context to avoid potential races of removing/adding netifs when iterating over the list
+    esp_netif_tcpip_exec(print_all_ips_tcpip, (void*) prefix);
+}
+
+
+esp_err_t net_connect(void)
+{
+#if CONFIG_NET_CONNECT_ETHERNET
+    ESP_LOGI(TAG, "Initializing Ethernet interface...");
+    if (net_connect_ethernet_connect() != ESP_OK) {
+        ESP_LOGE(TAG, "Failed to initialize Ethernet interface");
+        return ESP_FAIL;
+    }
+    ESP_ERROR_CHECK(esp_register_shutdown_handler(&net_connect_ethernet_shutdown));
+    ESP_LOGI(TAG, "Ethernet interface initialized successfully");
+#endif
+#if CONFIG_NET_CONNECT_WIFI
+    ESP_LOGI(TAG, "Initializing WiFi interface...");
+    if (!net_connect_wifi_is_configured()) {
+        /* Configure WiFi with Kconfig defaults */
+        if (net_configure_wifi_sta(NULL) == NULL) {
+            ESP_LOGE(TAG, "Failed to configure WiFi interface");
+            return ESP_FAIL;
+        }
+    }
+    /* Use new API for WiFi connection */
+    if (net_connect_wifi() != ESP_OK) {
+        ESP_LOGE(TAG, "Failed to initialize WiFi interface");
+        return ESP_FAIL;
+    }
+    ESP_ERROR_CHECK(esp_register_shutdown_handler(&net_connect_wifi_shutdown));
+    ESP_LOGI(TAG, "WiFi interface initialized successfully");
+#endif
+#if CONFIG_NET_CONNECT_THREAD
+    ESP_LOGI(TAG, "Initializing Thread interface...");
+    if (net_connect_thread_connect() != ESP_OK) {
+        ESP_LOGE(TAG, "Failed to initialize Thread interface");
+        return ESP_FAIL;
+    }
+    ESP_ERROR_CHECK(esp_register_shutdown_handler(&net_connect_thread_shutdown));
+    ESP_LOGI(TAG, "Thread interface initialized successfully");
+#endif
+#if CONFIG_NET_CONNECT_PPP
+    ESP_LOGI(TAG, "Initializing PPP interface...");
+    if (net_connect_ppp_connect() != ESP_OK) {
+        ESP_LOGE(TAG, "Failed to initialize PPP interface");
+        return ESP_FAIL;
+    }
+    ESP_ERROR_CHECK(esp_register_shutdown_handler(&net_connect_ppp_shutdown));
+    ESP_LOGI(TAG, "PPP interface initialized successfully");
+#endif
+
+#if CONFIG_NET_CONNECT_ETHERNET
+    net_connect_print_all_netif_ips(NET_CONNECT_NETIF_DESC_ETH);
+#endif
+
+#if CONFIG_NET_CONNECT_WIFI
+    net_connect_print_all_netif_ips(NET_CONNECT_NETIF_DESC_STA);
+#endif
+
+#if CONFIG_NET_CONNECT_THREAD
+    net_connect_print_all_netif_ips(NET_CONNECT_NETIF_DESC_THREAD);
+#endif
+
+#if CONFIG_NET_CONNECT_PPP
+    net_connect_print_all_netif_ips(NET_CONNECT_NETIF_DESC_PPP);
+#endif
+
+    return ESP_OK;
+}
+
+
+esp_err_t net_disconnect(void)
+{
+    ESP_LOGI(TAG, "Disconnecting network interfaces...");
+#if CONFIG_NET_CONNECT_ETHERNET
+    ESP_LOGI(TAG, "Deinitializing Ethernet interface...");
+    net_connect_ethernet_shutdown();
+    ESP_ERROR_CHECK(esp_unregister_shutdown_handler(&net_connect_ethernet_shutdown));
```

Comment:
Should we handle errors in more graceful way?

Applies to other places as well, uses `ESP_ERROR_CHECK()` in some places returns errors or error goto macros in others.

Reply:
Improved error handling (replacing ESP_ERROR_CHECK with proper error returns and cleanup paths) is planned for a future release. The current release intentionally avoids refactoring error-handling behavior.